deepspeed/module_inject/auto_tp.py

```diff
@@ -346,11 +346,15 @@ def _replace(self, child, name, conv_linear_layer):
                 weight, bias = shard_value_with_share_qk(child.weight.data, child.bias, dist.get_rank(),
                                                          dist.get_world_size(), False)
                 return LinearAllreduce(weight, bias, self.mp_group)
+        # For Arctic model, bypass to all_reduce replacement for w2 weights
+        arctic_w2_all_reduce_linear = False
+        if 'Arctic' in str(self.module) and 'w2' in name:
+            arctic_w2_all_reduce_linear = True
```

Comment:
Thanks for catching this :) May I ask why you don't add these layers to the all_reduce_linears rather than introducing a new argument here? I am afraid the logic here gets so messy with adding more and more options for specific models.

Reply:
I see, then in that case, I agree with your solution. Thanks @pi314ever for the clarification.